src/VisualStudio/CSharp/Impl/EventHookup/EventHookupSessionManager_EventHookupSession.cs

```diff
@@ -207,11 +220,14 @@ private IEventSymbol GetEventSymbol(SemanticModel semanticModel, SyntaxToken plu
                 return symbol as IEventSymbol;
             }
 
-            private string GetEventHandlerName(IEventSymbol eventSymbol, SyntaxToken plusEqualsToken, SemanticModel semanticModel, ISyntaxFactsService syntaxFactsService)
+            private string GetEventHandlerName(
+                IEventSymbol eventSymbol, SyntaxToken plusEqualsToken, SemanticModel semanticModel,
+                ISyntaxFactsService syntaxFactsService, NamingRule namingRule)
             {
                 AssertIsBackground();
-                var basename = string.Format("{0}_{1}", GetNameObjectPart(eventSymbol, plusEqualsToken, semanticModel, syntaxFactsService), eventSymbol.Name);
-                basename = basename.ToPascalCase(trimLeadingTypePrefix: false);
+                var objectPart = GetNameObjectPart(eventSymbol, plusEqualsToken, semanticModel, syntaxFactsService);
+                var basename = namingRule.NamingStyle.CreateName(ImmutableArray.Create(
```

Comment:
should there be an overload that just accepts a single word to avoid creating the array?